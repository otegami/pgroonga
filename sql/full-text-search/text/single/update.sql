CREATE TABLE memos (
  id integer,
  content text
);

INSERT INTO memos VALUES (1, 'PostgreSQL is a RDBMS.');
INSERT INTO memos VALUES (2, 'Groonga is fast full text search engine.');
INSERT INTO memos VALUES (3, 'PGroonga is a PostgreSQL extension that uses Groonga.');

CREATE INDEX grnindex ON memos USING pgroonga (content);

UPDATE memos SET content = 'Mroonga is a MySQL plugin that uses Groonga.'
 WHERE id = 3;

SET enable_seqscan = off;
SET enable_indexscan = on;
SET enable_bitmapscan = off;

SELECT id, content
  FROM memos
 WHERE content %% 'PostgreSQL';

SELECT id, content
  FROM memos
 WHERE content %% 'MySQL';

DROP TABLE memos;
