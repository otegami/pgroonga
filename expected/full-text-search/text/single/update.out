CREATE TABLE memos (
  id integer,
  content text
);
INSERT INTO memos VALUES (1, 'PostgreSQL is a RDBMS.');
INSERT INTO memos VALUES (2, 'Groonga is fast full text search engine.');
INSERT INTO memos VALUES (3, 'PGroonga is a PostgreSQL extension that uses Groonga.');
CREATE INDEX grnindex ON memos USING pgroonga (content);
UPDATE memos SET content = 'Mroonga is a MySQL plugin that uses Groonga.'
 WHERE id = 3;
SET enable_seqscan = off;
SET enable_indexscan = on;
SET enable_bitmapscan = off;
SELECT id, content
  FROM memos
 WHERE content %% 'PostgreSQL';
 id |        content         
----+------------------------
  1 | PostgreSQL is a RDBMS.
(1 row)

SELECT id, content
  FROM memos
 WHERE content %% 'MySQL';
 id |                   content                    
----+----------------------------------------------
  3 | Mroonga is a MySQL plugin that uses Groonga.
(1 row)

DROP TABLE memos;
